### PR TITLE
Fix new PhpStan errors with version 1.8.8

### PR DESCRIPTION
**Pull request by blankse**
New errors with version 1.8.8 https://github.com/phpstan/phpstan/releases/tag/1.8.8
```
 ------ ----------------------------------------------------------------------------------------- 
  Line   bundles/EcommerceFrameworkBundle/FilterService/FilterType/ElasticSearch/NumberRange.php  
 ------ ----------------------------------------------------------------------------------------- 
  50     Variable $value in empty() always exists and is not falsy.                               
 ------ ----------------------------------------------------------------------------------------- 

 ------ --------------------------------------------------------------------------- 
  Line   bundles/EcommerceFrameworkBundle/FilterService/FilterType/NumberRange.php  
 ------ --------------------------------------------------------------------------- 
  62     Variable $value in empty() always exists and is not falsy.                 
 ------ --------------------------------------------------------------------------- 

 ------ -------------------------------------------------------------------------------------------------------------------- 
  Line   lib/DataObject/GridColumnConfig/Operator/Concatenator.php                                                           
 ------ -------------------------------------------------------------------------------------------------------------------- 
  63     Call to function method_exists() with 0|0.0|''|'0'|array{}|false|null and 'isEmpty' will always evaluate to false.  
 ------ -------------------------------------------------------------------------------------------------------------------- 

 ------ ------------------------------------------------------------------------ 
  Line   models/DataObject/ClassDefinition/Data/Time.php                         
 ------ ------------------------------------------------------------------------ 
  108    Call to function is_string() with mixed will always evaluate to false.  
 ------ ------------------------------------------------------------------------ 

 ------ --------------------------------------------------------------- 
  Line   models/User.php                                                
 ------ --------------------------------------------------------------- 
  404    Variable $roles in empty() always exists and is always falsy.  
 ------ --------------------------------------------------------------- 

                                                                                                                        
 [ERROR] Found 5 errors
```

### CODE REVIEW

File: lib/DataObject/GridColumnConfig/Operator/Concatenator.php [hunk 1]
```diff
@@ -63,7 +63,11 @@ public function getLabeledValue($element)
 
             foreach ($childValues as $value) {
                 if (!$hasValue) {
-                    if (!empty($value) || (method_exists($value, 'isEmpty') && !$value->isEmpty())) {
+                    if (is_object($value) && method_exists($value, 'isEmpty')) {
+                        if (!$value->isEmpty()) {
+                            $hasValue = true;
+                        }
```
Comment (jdreesen) [inline lib/DataObject/GridColumnConfig/Operator/Concatenator.php]: Maybe we can write it like this:

```suggestion
                        $hasValue = !$value->isEmpty();
```

File: lib/DataObject/GridColumnConfig/Operator/Concatenator.php [hunk 2]
```diff
@@ -63,7 +63,9 @@ public function getLabeledValue($element)
 
             foreach ($childValues as $value) {
                 if (!$hasValue) {
-                    if (!empty($value) || (method_exists($value, 'isEmpty') && !$value->isEmpty())) {
+                    if (is_object($value) && method_exists($value, 'isEmpty')) {
+                        $hasValue = !$value->isEmpty();
+                    } elseif (!empty($value)) {
                         $hasValue = true;
```
Comment (blankse) [inline lib/DataObject/GridColumnConfig/Operator/Concatenator.php]:
```suggestion
                    } else {
                        $hasValue = !empty($value);
```

**Comment by mcop1** [comment]
@blankse thank you, you are amazing. Is the pr ready for review or do you want to make some more changes?

**Comment by blankse** [comment]
@mcop1 It is ready for review :)

**Comment by blankse** [comment]
@mcop1 Can you merge the 10.5 branch to 10.x branch? So the test there also passes?

**Pull request merged**